plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -250,11 +258,17 @@ public void handleEvent(){
 			String message = "Edit the port to be exposed by the service";
 			final IServicePort port = model.getSelectedServicePort();
 			final ServicePortAdapter target = new ServicePortAdapter(port);
-			ServicePortDialog dialog = new ServicePortDialog(target, message, model.getServicePorts());
+			final boolean previousRoutePort = model.getRoutingPort() == port;
+			ServicePortDialog dialog = new ServicePortDialog(target, message, model.getServicePorts(), previousRoutePort);
 			if(Window.OK == dialog.open()) {
 				target.setName(NLS.bind("{0}-tcp", target.getPort()));
 				model.updateServicePort(port, target);
 				model.setSelectedServicePort(target);
+				boolean routePort = dialog.isRoutePort();
+				if (routePort != previousRoutePort) {
+				    model.setRoutingPort(routePort?target:null);
+				    portsViewer.refresh();
```

Comment:
When you edit a port, you can change name, source port, target port and the flag that tells if this port is used by the route. If you edit only this last one, the port model is not changed and the UI won't be updated. Thus the refresh is we detect the flag has changed